### PR TITLE
rrdcached dump support

### LINKED ISSUE #1141: rrdtool dump not works with rrdcached
**Describe the bug**
Man page said that rrdcached is supported by dump command, but really not works.

**To Reproduce**
check if rrd exist:
```
$ rrdtool last spb-sdn-local2/uptime.rrd --daemon 127.0.0.1:42217
1637163600
```
try dump:
```
$ rrdtool dump spb-sdn-local2/uptime.rrd --daemon 127.0.0.1:42217
ERROR: opening 'spb-sdn-local2/uptime.rrd': No such file or directory
```
(dump without rrdcached is works):
```
$ rrdtool dump /opt/observium/rrd/spb-sdn-local2/uptime.rrd |head
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE rrd SYSTEM "https://oss.oetiker.ch/rrdtool/rrdtool.dtd">
<!-- Round Robin Database Dump -->
<rrd>
        <version>0003</version>
        <step>300</step> <!-- Seconds -->
        <lastupdate>1637163709</lastupdate> <!-- 2021-11-17 18:41:49 MSK -->
```

RRDtool on Ubuntu 20.04:
```
RRDtool 1.7.2  Copyright by Tobias Oetiker <tobi@oetiker.ch>
               Compiled Feb 26 2020 17:10:48
```

### PR DESCRIPTION
**Pull request by thz**
fixes #1141

### CODE REVIEW

File: src/rrd_client.c [hunk 1]
```diff
@@ -2059,6 +2060,125 @@
...
+            return -1;
+        }
+        if (received == 0) {
+            close_connection(client);
+            break; // EOF
+        }
+        written = output_cb(buffer, received, cb_userdata);
+        if (written != received) {
+            rrd_set_error("rrdc_dump: unexpected number of bytes (%d) "
+                    "written (output_cb)", written);
```
[the remark this answers is not in the document]
Comment (thz) [reply src/rrd_client.c:2147]: `%d` --> `%ld` fix pushed.

**Comment by thz** [comment]
The grep 404 was recently "recently" fixed upstream: https://github.com/microsoft/vcpkg/pull/29067

Would this need a bump of the `vcpkgCommitId` now?

**Comment by oetiker** [comment]
> The grep 404 was recently "recently" fixed upstream: [microsoft/vcpkg#29067](https://github.com/microsoft/vcpkg/pull/29067)
> 
> Would this need a bump of the `vcpkgCommitId` now?

@thz please try :)

**Comment by thz** [comment]
> > The grep 404 was recently "recently" fixed upstream: [microsoft/vcpkg#29067](https://github.com/microsoft/vcpkg/pull/29067)
> > Would this need a bump of the `vcpkgCommitId` now?
> 
> @thz please try :)

I just tested a manual vcpkg run from github.com/microsoft/vcpkg HEAD and it succeeded building the deps (including a slightly newer grep version). So I committed the SHA1 of the last tag as `vcpkgCommitId`...
I think CI needs approval again.

**Comment by c72578** [comment]
Here is some background info concerning the updated vcpkg, which is part of this PR:
- Update vcpkg to current [2023.10.19 Release](https://github.com/microsoft/vcpkg/releases/tag/2023.10.19), commit [8eb5735](https://github.com/microsoft/vcpkg/commit/8eb57355a4ffb410a2e94c07b4dca2dffbee8e50)
- Current versions of libraries are e.g.:
  cairo 1.17.8, expat 2.5.0, fontconfig 2.14.2, freetype 2.12.1,
  gettext 0.21.1, glib 2.78.0, libpng 1.6.39, pango 1.50.14,
  pcre2 10.40, libxml2 2.11.5 and zlib 1.3

**Comment by thz** [comment]
Oh nice. Ready for merge then.

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
thank you!

**Pull request merged**